### DISCUSSION TITLE
Possible to mix Feliz.React and Fable.Lit in a single app?

### DISCUSSION THREAD

**Discussion by leolorenzoluis** [fable-compiler/Fable.Lit#61] [Q&A]
There are great libraries written for Fable but leverages React. How do you mix two frameworks into a single one? Leveraging existing react controls in Feliz and also creating your own web components with Lit?

**Comment by alfonsogarciacaro** [comment]
There's actually a helper here to do that. Not sure if it's the best approach but it works: https://github.com/fable-compiler/Fable.Lit/blob/3e2930170a284a2bcbeb5cb37bade9e10b6aafb0/src/Lit.React/Lit.React.fs#L47-L65

You can get the `Fable.Feliz.Lit` package from Nuget or simply copy that code into your project. Please note you need to use in a React component, like:

```fsharp
open Feliz
open Lit

[<ReactComponent>]
let MyReactWrapper (aValue: int) =
    MyLitComponent aValue |> React.ofLit
```

**Reply by leolorenzoluis** [reply]
What about converting a react component to a lit? I believe your sample code converts an existing Lit component to React but not vice versa?

**Reply by alfonsogarciacaro** [reply]
There's a helper in the same file to do exactly that :) Although I'm not entirely sure if the solution has performance issues. https://github.com/fable-compiler/Fable.Lit/blob/3e2930170a284a2bcbeb5cb37bade9e10b6aafb0/src/Lit.React/Lit.React.fs#L42-L45

**Reply by leolorenzoluis** [reply]
@alfonsogarciacaro Could we leverage the `@lit-labs/react` instead? I tried using this helper but I'm faced with lifecycle bug issues (i'm guessing here)

EDIT: Ok I conclude mixing Lit and React doesn't play nice with its own lifecycles but again maybe I'm too tired mixing renders that don't play well.
